webui/react/src/components/TableFilterSearch.tsx

```diff
@@ -0,0 +1,66 @@
+import { Button, Input } from 'antd';
+import { FilterDropdownProps } from 'antd/es/table/interface';
+import React, { useCallback, useState } from 'react';
+
+import Icon from './Icon';
+import css from './TableFilterSearch.module.scss';
+
+interface Props extends FilterDropdownProps {
+  onReset?: () => void;
+  onSearch?: (search: string) => void;
+  value: string;
+}
+
+const TableFilterSearch: React.FC<Props> = ({
+  clearFilters,
+  confirm,
+  onReset,
+  onSearch,
+  value,
+}: Props) => {
+  const [ search, setSearch ] = useState(value);
+
+  const handleSearchChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearch(e.target.value || '');
+  }, []);
+
+  const handleReset = useCallback(() => {
+    setSearch('');
+    if (onReset) onReset();
+    if (clearFilters) clearFilters();
+  }, [ clearFilters, onReset ]);
+
+  const handleSearch = useCallback(() => {
+    if (onSearch) onSearch(search);
+    confirm();
+  }, [ confirm, onSearch, search ]);
+
+  return (
+    <div className={css.base}>
+      <div className={css.search}>
+        <Input
```

Comment:
another nit thing would be to make sure that the text input is auto-focused after clicking the search icon and making this component render?